src/components/project/project.service.ts

```diff
@@ -559,12 +546,9 @@ export class ProjectService {
     // we'll use this course all/nothing check. This, assuming role permissions
     // are set correctly, allows the users which can view all projects & their members
     // to use this feature.
-    const perms = await this.authorizationService.getPermissions({
-      resource: User,
-      sessionOrUserId: session,
-    });
+    const perms = this.privileges.for(session, User).all.projects;
 
-    if (!perms.projects.canRead) {
+    if (!perms.read) {
       return SecuredList.Redacted;
     }
```

Comment:
So really we can drop perms here alltogether then as this little check is the only reason it gets used anyway

Reply:
Actually with the comment above this, I think there is a broader issue with the DB like you indicated
```
    // Instead of trying to handle which subset of projects should be included,
    // based on doing the work of seeing which project teams they can view,
    // we'll use this course all/nothing check. This, assuming role permissions
    // are set correctly, allows the users which can view all projects & their members
    // to use this feature.
```

Reply:
> What's not covered in the DB query is whether the user can list project members. So if the can see the project but not it's members and then they filter on a certain user and a project, the correlation is leaked.
Let me see if I can adjust the DB query.

I looked into this. Currently all of our roles have the same perms for projects & project members. So the check currently in place is sufficient. Though it would be worth revisiting after DB migration.


---

As I've been scrolling through I've seen this pattern in several spots. So what I think we should do is leave this here as you have it. And then do another pass later, to ensure our all of our lists work with these "subsets" / conditional items.
That would be ensuring the DB queries reference privileges and removing the check from the service files.

Reply:
Added a ticket for this: https://seed-company-squad.monday.com/boards/3451697530/pulses/5008840819